### PR TITLE
Vgm vpc subseg

### PR DESCRIPTION
**Pull request by yifengli2023**
Similar PR as https://github.com/GoogleCloudPlatform/magic-modules/pull/8989, but with additional fixes in the Service Perimeters tests.

Moving changes for Access Context Manager's VPC Sub-segmentation feature from Terraform EAP to the normal provider.

[Here](https://cloud-internal-review.git.corp.google.com/c/cloud-graphite-eng/magic-modules-private-overrides/+/42350) is the PR that added these same fields to Terraform EAP.

[Here](https://source.corp.google.com/piper///depot/google3/google/identity/accesscontextmanager/v1/service_perimeter.proto;rcl=562860862;l=450-457) are the new fields for Service Perimeters, so this PR adds these new fields (and associated sub-fields) to various Service Perimeter-related tests and resource files.

Similarly, [here](https://source.corp.google.com/piper///depot/google3/google/identity/accesscontextmanager/v1/access_level.proto;rcl=562860862;l=164-166) is the new field for Access Levels. This PR adds this new field (and associated sub-fields) to various Access Level-related tests and resource files.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
accesscontextmanager: added support for Access Context Manager's VPC Sub-segmentation feature
```

### CODE REVIEW

File: mmv1/products/accesscontextmanager/AccessLevel.yaml [hunk 1]
```diff
@@ -243,6 +243,22 @@ properties:
...
+              description: 'The request must originate from one of the provided VPC networks in Google Cloud. Cannot specify this field together with `ip_subnetworks`.'
+              item_type: !ruby/object:Api::Type::NestedObject
+                properties:
+                  - !ruby/object:Api::Type::NestedObject
+                    name: 'vpcSubnetwork'
+                    description: 'Sub networks within a VPC network.'
+                    properties:
+                      - !ruby/object:Api::Type::String
+                        name: 'network'
+                        description: 'Required. Network name to be allowed by this Access Level. Networks of foreign organizations requires `compute.network.get` permission to be granted to caller.'
```
Comment (slevenick) [inline mmv1/products/accesscontextmanager/AccessLevel.yaml:257]: This should be required: true as noted in the decription

File: mmv1/products/accesscontextmanager/AccessLevelCondition.yaml [hunk 2]
```diff
@@ -212,3 +212,19 @@ properties:
...
+    name: 'vpcNetworkSources'
+    description: 'The request must originate from one of the provided VPC networks in Google Cloud. Cannot specify this field together with `ip_subnetworks`.'
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::NestedObject
+          name: 'vpcSubnetwork'
+          description: 'Sub networks within a VPC network.'
+          properties:
+            - !ruby/object:Api::Type::String
+              name: 'network'
```
Comment (slevenick) [inline mmv1/products/accesscontextmanager/AccessLevelCondition.yaml:225]: required: true

File: mmv1/products/accesscontextmanager/AccessLevels.yaml [hunk 3]
```diff
@@ -234,6 +234,22 @@ properties:
...
+                    name: 'vpcNetworkSources'
+                    description: 'The request must originate from one of the provided VPC networks in Google Cloud. Cannot specify this field together with `ip_subnetworks`.'
+                    item_type: !ruby/object:Api::Type::NestedObject
+                      properties:
+                        - !ruby/object:Api::Type::NestedObject
+                          name: 'vpcSubnetwork'
+                          description: 'Sub networks within a VPC network.'
+                          properties:
+                            - !ruby/object:Api::Type::String
+                              name: 'network'
```
Comment (slevenick) [inline mmv1/products/accesscontextmanager/AccessLevels.yaml:247]: required: true

**Review by slevenick — CHANGES_REQUESTED** [review]
Just a couple changes needed based on the description of the fields not matching the function.

Other than that, looks good to me, thanks for adding tests!

File: mmv1/products/accesscontextmanager/AccessLevel.yaml (hunk 1 above)
Comment (yifengli2023) [reply mmv1/products/accesscontextmanager/AccessLevel.yaml:258]: Done!

File: mmv1/products/accesscontextmanager/AccessLevelCondition.yaml (hunk 2 above)
Comment (yifengli2023) [reply mmv1/products/accesscontextmanager/AccessLevelCondition.yaml:225]: Done!

File: mmv1/products/accesscontextmanager/AccessLevels.yaml (hunk 3 above)
Comment (yifengli2023) [reply mmv1/products/accesscontextmanager/AccessLevels.yaml:247]: Done!

**Review by slevenick — APPROVED** [review]

**Pull request merged**